File: src/main/java/no/stunor/origo/eventorapi/model/origo/calendar/CalendarRace.java
package no.stunor.origo.eventorapi.model.origo.calendar;

import java.util.Date;
import java.util.List;
import java.util.Map;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import no.stunor.origo.eventorapi.model.firestore.Eventor;
import no.stunor.origo.eventorapi.model.firestore.Organisation;
import no.stunor.origo.eventorapi.model.origo.common.Position;
import no.stunor.origo.eventorapi.model.origo.entry.EntryBreak;
import no.stunor.origo.eventorapi.model.origo.event.DisiplineEnum;
import no.stunor.origo.eventorapi.model.origo.event.DistanceEnum;
import no.stunor.origo.eventorapi.model.origo.event.EventClassificationEnum;
import no.stunor.origo.eventorapi.model.origo.event.EventFormEnum;
import no.stunor.origo.eventorapi.model.origo.event.EventStatusEnum;
import no.stunor.origo.eventorapi.model.origo.event.LightConditionEnum;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CalendarRace{
    private Eventor eventor;
    private String eventId;
    private String raceId;
    private String eventName;
    private String raceName;
    private Date raceDate;
    private EventFormEnum type;
    private EventClassificationEnum classification;
    private LightConditionEnum lightCondition;
    private DistanceEnum distance;
    private Position position;
    private EventStatusEnum status;
    private List<DisiplineEnum> disciplines;
    private List<Organisation> organisers;
    private List<EntryBreak> entryBreaks;
    private boolean signedUp;
    private int entries;
    private Map<String,Integer> organisationEntries;
    private boolean startList;
    private boolean resultList;
    private boolean livelox;
}

File: src/main/java/no/stunor/origo/eventorapi/model/calendar/CalendarRace.kt
package no.stunor.origo.eventorapi.model.calendar

import com.google.cloud.Timestamp
import no.stunor.origo.eventorapi.model.event.*
import no.stunor.origo.eventorapi.model.organisation.Organisation
import no.stunor.origo.eventorapi.model.event.Position
import no.stunor.origo.eventorapi.model.event.EntryBreak
import java.io.Serializable

data class CalendarRace(
        var eventorId: String = "",
        var eventId: String = "",
        var eventName: String = "",
        var raceId: String = "",
        var raceName: String? = null,
        var raceDate: Timestamp? = null,
        var type: EventFormEnum = EventFormEnum.Individual,
        var classification: EventClassificationEnum = EventClassificationEnum.Club,
        var lightCondition: LightConditionEnum = LightConditionEnum.Day,
        var distance: DistanceEnum = DistanceEnum.Middle,
        var position: Position? = null,
        var status: EventStatusEnum = EventStatusEnum.Applied,
        var disciplines: List<DisciplineEnum> = listOf(),
        var organisers: List<Organisation> = listOf(),
        var entryBreaks: List<EntryBreak> = listOf(),
        var entries: Int = 0,
        var userEntries: MutableList<CalendarCompetitor> = mutableListOf(),
        var organisationEntries: MutableMap<String, Int> = mutableMapOf(),
        var signedUp: Boolean = false,
        var startList: Boolean = false,
        var resultList: Boolean = false,
        var livelox: Boolean = false
) : Serializable